### PR TITLE
Fix Implicitly Concatenated String Literals (Ruff ISC001)

### LINKED ISSUE #14910: [Refactor] Fix Implicitly Concatenated String Literals - Ruff Rule ISC001
There are multiple instances in the Astropy codebase where implicitly concatenated string literals are used. This coding style violates the coding guidelines and makes the code less readable. Implicit concatenation should be avoided in favor of explicit concatenation using the '+' operator or using parentheses to span the string literals across multiple lines.

Example:
astropy/convolution/tests/test_kernel_class.py:226:39: ISC001 Implicitly concatenated string literals on one line

```
|
226 | with pytest.warns(
227 | AstropyUserWarning, match=r"Both array and kernel " r"are Kernel instances"
| ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ ISC001
```

Related to: #14818

### PR DESCRIPTION
**Pull request by Maia-jp**
This pull request fixes a series of Implicitly Concatenated String Literals issues (ISC001).

The following files and lines have been addressed:
- astropy/convolution/tests/test_kernel_class.py:226:39
- astropy/convolution/tests/test_kernel_class.py:241:39
- astropy/convolution/tests/test_kernel_class.py:378:19
- astropy/convolution/tests/test_kernel_class.py:396:19
- astropy/coordinates/tests/test_iau_fullstack.py:206:15
- astropy/io/ascii/daophot.py:32:9
- astropy/io/ascii/daophot.py:32:17
- astropy/io/ascii/tests/test_ipac_definitions.py:101:15
- astropy/io/fits/hdu/image.py:275:25
- astropy/io/fits/tests/test_fitstime.py:392:19
- astropy/table/tests/test_column.py:762:15
- astropy/table/tests/test_column.py:775:15
- astropy/table/tests/test_column.py:815:15
- astropy/table/tests/test_table.py:2237:19
- astropy/table/tests/test_table.py:2672:15
- astropy/timeseries/periodograms/lombscargle_multiband/core.py:236:38
- astropy/units/tests/test_logarithmic.py:209:13
- astropy/units/tests/test_logarithmic.py:211:21

The changes involve explicitly concatenating the string literals

Fix #14910

### CODE REVIEW

**Review by nstarman — CHANGES_REQUESTED** [review]
I think these are all very uncontroversial. Most probably arose during the `black` PR. A nice cleanup!

@Maia-jp, please see the "Editing Ruff configuration" section of https://github.com/astropy/astropy/issues/14818#issue-1707345588. We want to ensure that this rule continues to be enforced.

**Review by eerovaher — COMMENTED** [review]
The changes to the code look good enough to merge, but I'd recommend squashing the commits and rebasing on latest `main` just to be sure that no new `ISC001` violations have been introduced since the pull request was opened.

**Review by nstarman — APPROVED** [review]
Thanks @Maia-jp !

**Comment by eerovaher** [comment]
The changes look good, but the CI image test failed for unrelated reasons and that's blocking the merge. Rebasing on current `main` should allow the CI tests to succeed, or maybe @pllim has a better idea on how to proceed.

**Comment by eerovaher** [comment]
I've rebased this and all the required tests have succeeded, so this should be good to merge.

**Comment by dhomeier** [comment]
> I think these are all very uncontroversial. Most probably arose during the `black` PR. A nice cleanup!

Yes, typically artefacts from `black` merging lines – is there some configuration available in `black` to prevent this, or are we just relying on `ruff` cleaning up after `black` from now on?

**Review by dhomeier — APPROVED** [review]
Good to go for `io.ascii`, thanks!

**Comment by nstarman** [comment]
> > I think these are all very uncontroversial. Most probably arose during the `black` PR. A nice cleanup!
> 
> 
> 
> Yes, typically artefacts from `black` merging lines – is there some configuration available in `black` to prevent this, or are we just relying on `ruff` cleaning up after `black` from now on?

I think it's actually a trickier problem since ruff runs before black (as is recommended). So if Black introduces one of these issues I don't think it's caught until a cron job runs. I don't think that should stop us from merging this PR, but we should think how to detect this in the future. Of course, if people have pre-commit installed on their local machine then pre-commit runs locally and then on GitHub and will catch any issue on ruff's second run.

**Comment by eerovaher** [comment]
> ...is there some configuration available in `black` to prevent this, or are we just relying on `ruff` cleaning up after `black` from now on?

Black has the `--experimental-string-processing` option that turns on [improved string processing](https://black.readthedocs.io/en/stable/the_black_code_style/future_style.html#improved-string-processing), but using it alone is deprecated. It is possible to use it with the [`--preview` option](https://black.readthedocs.io/en/stable/the_black_code_style/future_style.html#preview-style), but that also turns on a few more features. But [APE-20 specifies that `astropy` uses Black with its default settings](https://github.com/astropy/astropy-APEs/blob/main/APE20.rst#implementation). Letting Ruff check for ISC001 violations should work fine in practice and seems to be simpler than going through the process of updating the APE.

> ...So if Black introduces one of these issues I don't think it's caught until a cron job runs...

If Black edits the code then that causes the `pre-commit.ci` check to fail.

**Comment by nstarman** [comment]
> > ...So if Black introduces one of these issues I don't think it's caught until a cron job runs...

> If Black edits the code then that causes the pre-commit.ci check to fail.

True! wonderful.

**Comment by nstarman** [comment]
@taldcroft @saimn @larrybradley @WilliamJamieson @eerovaher are we good to merge?

**Comment by eerovaher** [comment]
I was the one who rebased this so I'd prefer not to review. But if I would've had any objections I would not have rebased.

**Comment by dhomeier** [comment]
> Black has the `--experimental-string-processing` option that turns on [improved string processing](https://black.readthedocs.io/en/stable/the_black_code_style/future_style.html#improved-string-processing), but using it alone is deprecated. It is possible to use it with the [`--preview` option](https://black.readthedocs.io/en/stable/the_black_code_style/future_style.html#preview-style), but that also turns on a few more features. But [APE-20 specifies that `astropy` uses Black with its default settings](https://github.com/astropy/astropy-APEs/blob/main/APE20.rst#implementation). Letting Ruff check for ISC001 violations should work fine in practice and seems to be simpler than going through the process of updating the APE.
> > ...So if Black introduces one of these issues I don't think it's caught until a cron job runs...
> 
> If Black edits the code then that causes the `pre-commit.ci` check to fail.

🤯 And there is no APE for Ruff usage... Well, hopefully broken string concatenations in new commits should be easily enough spotted; and once merged into single strings, black would leave them alone, right?

**Review by taldcroft — APPROVED** [review]
LGTM

**Comment by nstarman** [comment]
> once merged into single strings, black would leave them alone, right?

It won't break up a string, but it might move the whole string to a new line.

**Comment by nstarman** [comment]
We now have three explicit, and one implicit approval, so I'll merge by end of day unless someone has an objection

**Review by dhomeier — APPROVED** [review]
Did not find any new broken strings in `main` since rebasing either.

**Pull request merged**
